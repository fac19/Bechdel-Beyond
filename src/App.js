import React from 'react';
import {
	BrowserRouter as Router,
	Switch,
	Route,
	Redirect,
} from 'react-router-dom';

import { makeStyles } from '@material-ui/core/styles';

import MenuAppBar from './components/MenuAppBar';
import AllFilms from './components/Home/AllFilms';
import SignUp from './components/UserTasks/Signup/Signup';
import MoviePage from './components/Movie/MoviePage';
import LogIn from './components/UserTasks/Login/Login';

// import AboutPage from './components/About/AboutPage';
import ReviewPage from './components/Review/ReviewPage';
// import UserPage from './components/UserTasks/UserPage/UserPage';

import getData from './utils/getData';

const useStyles = makeStyles((theme) => ({
	root: {
		marginTop: theme.spacing(8),
	},
}));
function App() {
	const classes = useStyles();
	const [filmData, setFilmData] = React.useState('');

	const accessToken = window.localStorage.getItem('access token');
	const [loggedIn, setLoggedIn] = React.useState(null);
	const [currentPage, setCurrentPage] = React.useState('HOME');

	React.useEffect(() => {
		getData('/films').then((data) => setFilmData(data));
	}, []);
	return (
		<Router>
			<div className="App">
				<header className={`App-header ${classes.root}`}>
					<MenuAppBar
						setCurrentPage={setCurrentPage}
						currentPage={currentPage}
						loggedIn={loggedIn}
						setLoggedIn={setLoggedIn}
						accessToken={accessToken}
					/>
				</header>
				<Switch>
					<Route
						exact
						path="/"
						component={() => (
							<AllFilms filmData={filmData} setFilmData={setFilmData} />
						)}
					></Route>
					<Route
						exact
						path="/signup"
						render={() =>
							loggedIn ? (
								<Redirect to="/" />
							) : (
								<SignUp setLoggedIn={setLoggedIn} />
							)
						}
					/>

					<Route
						exact
						path="/login"
						render={() =>
							loggedIn ? (
								<Redirect to="/" />
							) : (
								<LogIn setLoggedIn={setLoggedIn} />
							)
						}
					/>

					<Route
						exact
						path="/create-review/:title"
						render={(props) =>
							loggedIn ? (
								<ReviewPage {...props} filmData={filmData} />
							) : (
								<LogIn setLoggedIn={setLoggedIn} />
							)
						}
					></Route>
					{/* {accessToken ? (
            <Route exact path="/user-profile" component={UserPage}></Route>
          ) : (
            ''
          )} */}
					{/*<Route exact path="/about" component={AboutPage}></Route> */}
					<Route exact path="/movie/:title" component={MoviePage}></Route>
					<Route path="*" render={() => <Redirect to="/" />} />
				</Switch>
			</div>
		</Router>
	);
}

export default App;
