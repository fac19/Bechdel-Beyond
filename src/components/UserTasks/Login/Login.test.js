import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import LogIn from './Login';
import App from '../../../App';

const setup = () => {
	const login = render(
		<Router>
			<LogIn />
		</Router>,
	);
	const emailInput = screen.getByLabelText(/Email/i, { selector: 'input' });
	const passwordInput = login.getByLabelText(/Password/i, {
		selector: 'input',
	});

	return {
		emailInput,
		passwordInput,
		...login,
	};
};

const renderHome = () => {
	const homePage = render(<App />);
	return { ...homePage };
};

test('Check login tests are running', () => {
	expect(true).toBeTruthy();
});

test('Check that Login input form displays', () => {
	render(
		<Router>
			<LogIn />
		</Router>,
	);
	screen.getByText('Email Address');
	screen.getByText('Password');
});

test('Can fill in input boxes and login', () => {
	const { emailInput, passwordInput } = setup();
	expect(emailInput.value).toBe('');
	fireEvent.change(emailInput, { target: { value: 'guy@someguy.com' } });
	fireEvent.change(passwordInput, { target: { value: '123' } });
	expect(emailInput.value).toBe('guy@someguy.com');
	const LoginBtn = screen.getByText('Sign In');
	fireEvent.click(LoginBtn);
});

it('Login page renders correctly', () => {
	const login = renderer
		.create(
			<Router>
				<LogIn />
			</Router>,
		)
		.toJSON();
	expect(login).toMatchSnapshot();
});
