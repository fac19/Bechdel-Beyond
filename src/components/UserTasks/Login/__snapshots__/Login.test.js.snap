// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-547"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-548 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <form
      className="makeStyles-form-549"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="email"
          id="email-label"
        >
          Email Address
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="email"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="email"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-663 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-665"
            >
              <span>
                Email Address
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-663 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-665"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        
      </p>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-550 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Sign In
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Forgot password?
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="/signup"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Don't have an account? Sign Up
          </a>
        </div>
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root MuiBox-root-809"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://material-ui.com/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Your Website
      </a>
       
      2020
      .
    </p>
  </div>
</main>
`;
